src/routes/reset-password/reset-request.ts

```diff
@@ -0,0 +1,78 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { knexController } from '../../database/database';
+import {
+  passwordResetRequestBody,
+  PasswordResetRequestBody,
+  passwordResetResponseBody,
+} from '../../types/auth.types';
+import { log } from '../../lib/utils/log';
+import { redisClient } from '../../lib/auth/redis';
+import { v4 as uuid } from 'uuid';
+import { hashPassword } from '../../lib/auth/auth';
+import { sendEmail } from '../../lib/utils/sendEmail';
+
+export const PASSWORD_RESET_TOKEN_EXPIRE_TIME = 600; // ten minutes
+
+const schema = {
+  tags: ['Auth'],
+  description: 'Responds with 202 Accepted regardless of unverified email.',
+  body: passwordResetRequestBody,
+  response: {
+    202: passwordResetResponseBody,
+    400: { $ref: 'error' },
+  },
+};
+
+schema.body.properties.email.example = 'john.doe@example.com';
+
+const handler = async (
+  request: FastifyRequest<{ Body: PasswordResetRequestBody }>,
+  reply: FastifyReply
+): Promise<void> => {
+  await reply
+    .code(202)
+    .send({ message: 'Password reset email is being sent.' });
+
+  const userInfo: { id: string; email: string } | undefined =
+    await knexController('user')
+      .where('email', request.body.email)
+      .where('archived_at', null)
+      .first('id', 'email');
+
+  if (userInfo === undefined) {
+    log.info('password reset request with non existent email');
+    return;
+  }
```

Comment:
404?

Reply:
Sillä voi haarukoida palvelun käyttäjiä joten eii

Reply:
Haarukointia voi harrastaa myös koitamalla rekistöröityä palveluun samalla säpolla että...

Reply:
Onko niin voittoisaa että tällä endpointilla ei voi sitä tehdä vs käyttäjäkokemus rajapinnalle? Monethan tuota tekee kyllä, että ei nimenomaan tällä endpointilla anna haarukoida, mutta en tiiä lisääkö se kenenkään turvallisuutta kauheasti.

Järjestelmään olis hyvä lisätä CAPTCHA tjsp jotta käyttäjiä ei voi kuitenkaan koneellisesti hakea rekistöröityessä / salasanaa resetoidessa.

Reply:
[Owasp lienee tässä nyt se auktoriteetti:](https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html#introduction)

- Return a consistent message for both existent and non-existent accounts.
- Ensure that the time taken for the user response message is uniform.

Reply:
Juu noudatetaan vaan. Kunhan horisen